BibBuddy/app/src/main/java/de/bibbuddy/BookFragment.java

```diff
@@ -127,21 +129,80 @@ public boolean onOptionsItemSelected(MenuItem item) {
 
     return super.onOptionsItemSelected(item);
   }
+      
+  private void handleChangeBookData() {
+    if (selectedBookItems.isEmpty()) {
+      return;
+    }
+
+    Bundle bundle = new Bundle();
+    BookItem bookItem = selectedBookItems.get(0);
+    bundle.putString(LibraryKeys.SHELF_NAME, shelfName);
+    bundle.putLong(LibraryKeys.SHELF_ID, shelfId);
+    bundle.putLong(LibraryKeys.BOOK_ID, bookItem.getId());
+
+    BookFormFragment bookFormFragment = new BookFormFragment(this);
+
+    bookFormFragment.setArguments(bundle);
+    getActivity().getSupportFragmentManager().beginTransaction()
+        .replace(R.id.fragment_container_view, bookFormFragment, LibraryKeys.FRAGMENT_BOOK)
+        .addToBackStack(null)
+        .commit();
+  }
+
+  @Override
+  public void onBookChanged(Book book, List<Author> authorList) {
+    bookModel.updateBook(book, authorList);
+
+    adapter.setBookList(bookModel.getBookList(shelfId));
+    adapter.notifyDataSetChanged();
+  }
+
+  private void handleDeleteBook() {
+    AlertDialog.Builder alertDeleteBook = new AlertDialog.Builder(context);
+
+    alertDeleteBook.setCancelable(false);
+    alertDeleteBook.setTitle(R.string.delete_book);
+    alertDeleteBook.setMessage(R.string.delete_book_message);
+
+    alertDeleteBook.setNegativeButton(R.string.back, new DialogInterface.OnClickListener() {
+      @Override
+      public void onClick(DialogInterface dialog, int which) {
+      }
+    });
+
+    alertDeleteBook.setPositiveButton(R.string.delete, new DialogInterface.OnClickListener() {
+      @Override
+      public void onClick(DialogInterface dialog, int which) {
+        bookModel.deleteBooks(selectedBookItems, shelfId);
+
+        adapter.setBookList(bookModel.getCurrentBookList());
+        adapter.notifyDataSetChanged();
+
+        updateEmptyView(bookModel.getCurrentBookList());
+        Toast.makeText(context, getString(R.string.deleted_book), Toast.LENGTH_SHORT).show();
+        unselectBookItems();
+      }
+    });
 
-  private void checkEmptyBookList() {
-    DatabaseHelper dbHelper = new DatabaseHelper(getContext());
-    BookDao bd = new BookDao(dbHelper);
-    if (bd.getAllBooksForShelf(shelfId).isEmpty()) {
-      AlertDialog.Builder ee = new AlertDialog.Builder(getContext());
-      ee.setTitle(R.string.empty_shelf);
-      ee.setMessage(R.string.empty_shelf_description);
-      ee.setPositiveButton(R.string.ok,
+    alertDeleteBook.show();
+  }
+
+  private void checkEmptyShelf() {
+    dbHelper = new DatabaseHelper(getContext());
+    bookDao = new BookDao(dbHelper);
+
+    if (bookDao.getAllBooksForShelf(shelfId).isEmpty()) {
+      AlertDialog.Builder alertDialogEmptyShelf = new AlertDialog.Builder(getContext());
+      alertDialogEmptyShelf.setTitle(R.string.empty_shelf);
+      alertDialogEmptyShelf.setMessage(R.string.empty_shelf_description);
+      alertDialogEmptyShelf.setPositiveButton(R.string.ok,
```

Comment:
Please add a newline for better readability.
```suggestion

      alertDialogEmptyShelf.setPositiveButton(R.string.ok,
```